### PR TITLE
Refine ORTTrainer docstring descriptions

### PR DESCRIPTION
**Pull request by natke**
**Description**: Refine docstrings for ORTTrainer

**Motivation and Context**
- Adds clarity to descriptions and makes them more readable.

### CODE REVIEW

File: orttraining/orttraining/python/ort_trainer.py [hunk 1]
```diff
@@ -486,38 +486,65 @@ def __init__(self, model, loss_fn, model_desc, training_optimizer_name, map_opti
...
+               - a combined PyTorch model and loss function.
+                  Inputs to this combined PyTorch model are a concatenation of the
+                  model's input and the loss function's label input.
+                  Outputs are a concatenation of the loss function's output and the
+                  model's output.
+               - a combined ONNX model and loss function.
+        :param loss_fn: one of
+               - a PyTorch loss function if 'model' is a PyTorch model. A loss
+                 function takes two inputs (prediction, label) and output a loss
+                 tensor.
```
Comment (kit1980) [inline orttraining/orttraining/python/ort_trainer.py:502]: output_s_

Comment (kit1980) [inline orttraining/orttraining/python/ort_trainer.py:502]: Is the loss function really outputs a loss tensor? I think it outputs a loss scalar.

**Comment by thiagocrepaldi** [comment]
onnxruntime pytorch frontend doesn't use numpy formatting but Google's

File: orttraining/orttraining/python/ort_trainer.py [hunk 2]
```diff
@@ -486,38 +486,65 @@ def __init__(self, model, loss_fn, model_desc, training_optimizer_name, map_opti
...
+        :param model: one of
+               - a PyTorch model (class that inherits from torch.nn.Module)
+               - a combined PyTorch model and loss function.
+                  Inputs to this combined PyTorch model are a concatenation of the
+                  model's input and the loss function's label input.
+                  Outputs are a concatenation of the loss function's output and the
+                  model's output.
+               - a combined ONNX model and loss function.
+        :param loss_fn: one of
+               - a PyTorch loss function if 'model' is a PyTorch model. A loss
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/ort_trainer.py:500]: worth mentioning that the loss function must be a `torch.nn.Module`

File: orttraining/orttraining/python/ort_trainer.py [hunk 3]
```diff
@@ -486,38 +486,65 @@ def __init__(self, model, loss_fn, model_desc, training_optimizer_name, map_opti
...
+                 function takes two inputs (prediction, label) and output a loss
+                 tensor.
+               - None if model is already combined with a loss function.
+        :param model_desc: Specify input/output shapes, types, and names.
+               Must be consistent with the training model.
+        :param training_optimizer_name: one of
+               - 'SGDOptimizer'
+               - 'AdamOptimizer'
+               - 'LambOptimizer'
+        :param map_optimizer_attributes: for optimizers with weight-dependent
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/ort_trainer.py:510]: map_optimizer_attributes must be a functor (callable)

File: orttraining/orttraining/python/ort_trainer.py (hunk 1 above)
Comment (natke) [reply orttraining/orttraining/python/ort_trainer.py:502]: I think it is a tensor

File: orttraining/orttraining/python/ort_trainer.py (hunk 2 above)
Comment (natke) [reply orttraining/orttraining/python/ort_trainer.py:500]: Is that only when it is part of the combined model?

Comment (thiagocrepaldi) [reply orttraining/orttraining/python/ort_trainer.py:502]: loss_fn is either None (when it is present in the model) or torch.nn.Module, when it will be combined into the model and exported to ONNX. In fact, only torch.nn.Module can be exported into ONNX
